src/trace.rs

```diff
@@ -0,0 +1,67 @@
+#[derive(Debug, PartialEq)]
+pub enum TraceType {
+    Conservative,
+    Custom,
+}
+
+/// The trait used by the collector to determine how an object's fields should
+/// be traced.
+pub trait Trace {
+    fn trace(&self, _: &mut Vec<usize>) -> TraceType {
```

Comment:
Why is the argument `&mut Vec<usize>`? Should it be `&[u8]`?

Reply:
Because we push pointers into it inside the `trace` method.

Reply:
Ah!